### DISCUSSION TITLE
api to return in local server time zone

### DISCUSSION THREAD

**Discussion by kenlo888** [datalust/seq-tickets#1475] [Help] [answered]
Hi, I'm currently using the seq api in c# to pull out logs and results returning back in json.

var resultSet = await connection.Events.ListAsync(filter: defaultFilter, render: true, count: count, signal: SignalExpressionPart.Signal(apiFilterSignal) );

The results coming back I wanted to default timestamp to be returning back with the local server timestamp.

0: {timestamp: "2022-02-07T14:03:43.2991003Z",…}

Is there an option I can set as part of the API call to get the timestamp using the local server timestamp?  Thank you very much for your help.

**Comment by KodrAus** [comment, marked as answer]
Hey @kenlo888 :wave:

Seq doesn't have a parameter in its API to convert timestamps before returning them, but it will give you timestamps back with a UTC offset in ISO-8601 format to keep everything consistent across all possible consumers. You should be able to convert them into a specific timezone after getting the response back from the API using `DateTimeOffset` or [Noda Time](https://nodatime.org/).

If you'd instead like to make sure all events are ingested using the server's clock instead of the client's you can [use API keys to force events to use the server clock when ingested](https://docs.datalust.co/docs/api-keys#server-timestamps). This won't change the timestamp offset in the API response, but will potentially change the value of the timestamps being stored.

**Comment by kenlo888** [comment]
Thank you so much @KodrAus for pointing me to the right direction.  I'll work on converting the time when I receive the data back from the API with your suggestions.  Thanks!
